rotkehlchen/data/importer.py

```diff
@@ -1435,3 +1459,123 @@ def import_bisq_trades_csv(self, filepath: Path, **kwargs: Any) -> Tuple[bool, s
                 except KeyError as e:
                     return False, str(e)
         return True, ''
+
+    @staticmethod
+    def _group_binance_rows(
+        rows: List[BinanceCsvRow],
+        **kwargs: Dict[str, Any],
+    ) -> Tuple[int, Dict[Timestamp, List[BinanceCsvRow]]]:
+        multirows: Dict[Timestamp, List[BinanceCsvRow]] = defaultdict(list)
+        skipped_count = 0
+        formatstr = kwargs.get('timestamp_format')
+        for csv_row in rows:
+            try:
+                timestamp = deserialize_timestamp_from_date(
+                    date=csv_row['UTC_Time'],
+                    formatstr=formatstr if formatstr is not None else '%Y-%m-%d %H:%M:%S',  # type: ignore  # noqa: E501
+                    location='binance',
+                )
+                csv_row['Coin'] = asset_from_binance(csv_row['Coin'])
+                csv_row['Change'] = deserialize_asset_amount(csv_row['Change'])
+                del csv_row['UTC_Time']
+                del csv_row['User_ID']
+                del csv_row['Account']
+                del csv_row['Remark']
+                multirows[timestamp].append(csv_row)
+            except (DeserializationError, UnknownAsset) as e:
+                log.warning(f'Skipped binance csv row {csv_row} because of {str(e)}')
+                skipped_count += 1
+            except KeyError as e:
+                log.error(f'Malformed binance csv columns! Broke on row {csv_row}. {str(e)}')
+                return len(rows), {}
+
+        return skipped_count, multirows
+
+    @staticmethod
+    def _process_single_binance_sentries(
+        timestamp: Timestamp,
+        rows: List[BinanceCsvRow],
+    ) -> Sequence[Tuple[Type[BinanceSingleEntry], BinanceCsvRow]]:
+        processed: List[Tuple[Type[BinanceSingleEntry], BinanceCsvRow]] = []
+        for single_entry_class in SINGLE_BINANCE_ENTRIES:
+            for row in rows:
+                if single_entry_class.is_entry(row['Operation']):
+                    single_entry_class.process_entry(timestamp=timestamp, data=row)
+                    processed.append((single_entry_class, row))
+        return processed
+
+    @staticmethod
+    def _process_multiple_binance_entries(
+        timestamp: Timestamp,
+        rows: List[BinanceCsvRow],
+    ) -> Tuple[Optional[Type[BinanceEntry]], int]:
+        for multiple_entry_class in MULTIPLE_BINANCE_ENTRIES:
+            if multiple_entry_class.are_entries([row['Operation'] for row in rows]):
+                processed_count = multiple_entry_class.process_entries(
+                    timestamp=timestamp,
+                    data=rows,
+                )
+                return multiple_entry_class, processed_count
+        return None, 0
+
+    def _process_binance_rows(self, multi: Dict[Timestamp, List[BinanceCsvRow]]) -> None:
+        BinanceEntry.db = self.db
```

Comment:
Hehe no. As I wrote in the BinanceEntry class this is not a way to initialize a python class's variables.

User proper initialization. Can't be setting the user DB as a static element of the class. This is extremely brittle, wrong, error prone and will 100% come and bite us back in the future.